apps/music/elements/music-search-results.js

```diff
@@ -59,105 +58,99 @@ var template =
 </style>
 <gaia-fast-list id="list">
   <template>
-    <li>
-      <a href="\${url}" data-file-path="\${name}" data-section="\${section}">
-        <img style="visibility: \${name ? 'visible' : 'hidden'}">
-        <h3>\${title}</h3>
-        <p>\${subtitle}</p>
-      </a>
-    </li>
+    <a href="\${url}" data-file-path="\${name}" data-section="\${section}">
+      <div class="image"><img></div>
+      <h3>\${title}</h3>
+      <p>\${subtitle}</p>
+    </a>
   </template>
 </gaia-fast-list>`;
 
 proto.createdCallback = function() {
-  var shadowRoot = this.createShadowRoot();
-  shadowRoot.innerHTML = template;
-
-  var $id = shadowRoot.getElementById.bind(shadowRoot);
+  this.innerHTML = template;
 
   this.els = {
-    list: $id('list')
+    list: this.querySelector('#list')
   };
 
   this.els.list.configure({
-    getSectionName: (item) => {
-      return item.section;
-    },
-
-    getItemImageSrc: (item) => {
-      return this.getItemImageSrc(item);
-    }
+    getSectionName: (item) => item.section,
+    getItemImageSrc: (item) => this.getItemImageSrc(item)
   });
 
-  this.els.list.addEventListener('click', (evt) => {
-    var link = evt.target.closest('a');
-    if (link) {
-      evt.preventDefault();
+  this.els.list.addEventListener('click', (e) => this.onLinkClick(e));
+};
 
-      this.close().then(() => {
-        this.dispatchEvent(new CustomEvent('resultclick', {
-          detail: link
-        }));
-      });
-    }
+proto.attachedCallback = function() {
+  setTimeout(() => this.enableTransitions());
+};
+
+proto.enableTransitions = function() {
+  this.classList.add('transitions-on');
+};
+
+proto.onLinkClick = function(evt) {
+  var link = evt.target.closest('a');
+  if (!link) return;
+  evt.preventDefault();
+
+  this.close().then(() => {
+    this.emit('resultclick', link);
   });
 };
 
+// To be overriden by view
+// if images are required
 proto.getItemImageSrc = function() {};
 
 proto.open = function() {
   return new Promise((resolve) => {
-    window.requestAnimationFrame(() => {
+    requestAnimationFrame(() => {
       this.setAttribute('active', true);
       resolve();
     });
 
-    this.dispatchEvent(new CustomEvent('open'));
+    this.emit('open');
   });
 };
 
 proto.close = function() {
   return new Promise((resolve) => {
-    window.requestAnimationFrame(() => {
+    requestAnimationFrame(() => {
       this.removeAttribute('active');
+      this.clearResults();
       resolve();
     });
 
-    this.dispatchEvent(new CustomEvent('close'));
+    this.emit('close');
   });
 };
 
 proto.setResults = function(results) {
   if (results.length === 0) {
-    document.l10n.formatValue('search-no-result').then((noResult) => {
-      this.els.list.model = [{
-        title: noResult,
-        subtitle: ''
-      }];
-    });
-
-    return;
+    return document.l10n.formatValue('search-no-result')
```

Comment:
nit: I prefer leaving this the way it was :-)

`setModel()` has no return value, but `formatValue()` does, so our `setResults()` should not be conditionally returning a value.